census/estimator/trainer/task.py

```diff
@@ -0,0 +1,180 @@
+  
+  See command line help text for description of args.
+  Returns:
+    A function (output_dir) -> Experiment where output_dir is a string
+    representing the location of summaries, checkpoints, and exports.
+    this function is used by learn_runner to create an Experiment which
+    executes model code provided in the form of an Estimator and
+    input functions.
+
+    All listed arguments in the outer function are used to create an
+    Estimator, and input functions (training, evaluation, serving). 
+    Unlisted args are passed through to Experiment.
+  """
+  def _experiment_fn(output_dir):
+    # num_epochs can control duration if train_steps isn't
+    # passed to Experiment
+    train_input = model.generate_input_fn(
+        train_files,
+        num_epochs=num_epochs,
+        batch_size=train_batch_size,
+    )
+    # Don't shuffle evaluation data
+    eval_input = model.generate_input_fn(
+        eval_files,
+        batch_size=eval_batch_size,
+        shuffle=False
+    )
+    return tf.contrib.learn.Experiment(
+        model.build_estimator(
+            job_dir,
+            embedding_size=embedding_size,
+            # Construct layers sizes with exponetial decay
+            hidden_units=[
+                max(2, int(first_layer_size * scale_factor**i))
+                for i in range(num_layers)
+            ]
+        ),
+        train_input_fn=train_input,
+        eval_input_fn=eval_input,
+        # export strategies control the prediction graph structure
+        # of exported binaries.
+        export_strategies=[saved_model_export_utils.make_export_strategy(
+            model.serving_input_fn,
+            default_output_alternative_key=None,
+            exports_to_keep=1
+        )],
+        **experiment_args
+    )
+  return _experiment_fn
+
+
+if __name__ == '__main__':
+  parser = argparse.ArgumentParser()
+  # Input Arguments
+  parser.add_argument(
+      '--train-files',
+      help='GCS or local paths to training data',
+      nargs='+',
+      required=True
+  )
+  parser.add_argument(
+      '--num-epochs',
+      help="""\
+      Maximum number of training data epochs on which to train.
+      If both --max-steps and --num-epochs are specified,
+      the training job will run for --max-steps or --num-epochs,
+      whichever occurs first. If unspecified will run for --max-steps.\
+      """,
+      type=int,
+  )
+  parser.add_argument(
+      '--train-batch-size',
+      help='Batch size for training steps',
+      type=int,
+      default=40
+  )
+  parser.add_argument(
+      '--eval-batch-size',
+      help='Batch size for evaluation steps',
+      type=int,
+      default=40
+  )
+  parser.add_argument(
+      '--train-steps',
+      help="""\
+      Steps to run the training job for. If --num-epochs is not specified,
+      this must be. Otherwise the training job will run indefinitely.\
+      """,
+      type=int
+  )
+  parser.add_argument(
```

Comment:
I hate how tf.learn deals with eval. But there are some fun facts that the user should know.

Because eval is read with num_epochs=None:

1) if eval-batch-size *  eval-steps is larger than the eval data, eval will contain multiple copies of the eval data. It is is less, eval is not ran on the whole data set. 

2) if eval-batch-size * eval-steps does not perfectly divide the dataset, each eval will use a slightly different set of eval instances, so every eval event are not 100% comparable!

So this brings up the question: why don't we always just do 1 pass over the eval set, like how any other normal system would do? Answer: I don't think tf.learn lets us. But this should be documented in the help str of these args and/or the readme